x-pack/plugin/sql/qa/server/src/main/resources/agg.sql-spec

```diff
@@ -89,6 +89,8 @@ groupByConstantLocalWithWhereContradiction
 SELECT COUNT(*), 'c' c WHERE 1 = 2 GROUP BY c;
 groupByConstantWithHavingContradiction
 SELECT COUNT(*), 'c' c FROM test_emp GROUP BY c HAVING c = 'z';
+groupByConstantWithMultipleMatchingRecords
```

Comment:
This is a good test, but you need to add significantly more queries.  Just an example test query: `select 1 as c, 2 as d, 'abc', 'e' from test_emp where gender = 'F' AND false group by c, d, 'x' order by 2, max(salary) desc limit 3`. Search for other examples in the existent files and come up with other scenarios for test.

Reply:
Yes, we definitely need more tests here. There are a few more in #84574 (very similar to the query you propose) and I can port them easily, but unfortunately this fix is not sufficient to cover all of them, so some still fail.

Also thank you for the hint about `terminate_after`, I'm investigating it right now